crates/contracts/solidity/SimulateCode.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import { IStorageAccessible } from "./interfaces/IStorageAccessible.sol";
+
+/// @title A contract for simulating arbitrary code in the context of a
+/// `StorageAccessible` contract.
+contract SimulateCode {
+    /// @dev This looks like a constructor but it is not... In fact, nodes
+    /// support `eth_call`s for contract creation and **return the code of the
+    /// contract that would be created**. This means we can use contructors to
```

Comment:
```suggestion
    /// contract that would be created**. This means we can use constructors to
```

Reply:
Interesting - my spell-check nvim plugin isn't working for Solidity files 🤔.